packages/moonstone/Icon/Icon.js

```diff
@@ -36,6 +37,14 @@ const IconBase = kind({
 	name: 'Icon',
 
 	propTypes: /** @lends moonstone/Icon.IconBase.prototype */ {
+		/**
+		 * Flip the icon horizontally or vertically.
+		 *
+		 * @type {('horizontal'|'vertical')}
+		 * @public
+		 */
+		flip: PropTypes.string,
```

Comment:
I'd suggest we place this prop in `ui` as well for congruency with `ui/IconButton` adding it. I _think_ it makes sense to move the CSS there as well but I could be convinced that flipping should be theme-specific if you have an example in mind.